### ISSUE TITLE
Unhandled exception

### ISSUE THREAD

**Issue by alranel** [GSGBen/t2md#39]
Hello!
When running:

```
./t2md --output-folder output --no-numbering --single-file --remove-empty-folders
```

during phase 1, after many "Finished xxx", it crashes with this error:

```
Unhandled exception. System.IO.IOException: The process cannot access the file '/Users/alranel/Downloads/output/osx-x64/output/t2md/xxx.json' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Init(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.File.Create(String path)
   at GoldenSyrupGames.T2MD.Cli.ProcessTrelloBoardAsync(TrelloApiBoardModel trelloApiBoard, CliOptions options)
   at GoldenSyrupGames.T2MD.Cli.RunAsync(CliOptions options)
   at GoldenSyrupGames.T2MD.Cli.Main(String[] args)
   at GoldenSyrupGames.T2MD.Cli.<Main>(String[] args)
```

**Comment by GSGBen** [comment]
Thanks for the report!

- Did you have the `/Users/alranel/Downloads/output/osx-x64/output/t2md/` folder or `/Users/alranel/Downloads/output/osx-x64/output/t2md/xxx.json` path open in any other application at the same time?
- Does it happen every time you run that, or just intermittently?
- Does it throw on the same `.json` file every time, or different?
- I don't have a Mac to test on. Do you have a linux box you can run it on, and does it occur there too?

**Comment by alranel** [comment]
Thank you for the quick reply :)
Here's the answers to your question:

* Yes! I do have two boards with the same name.
* No, I did not have the folder or any other file open in other applications.
* It was happening consistently.
* I can't reproduce it anymore because now the process is failing with the error reported in #42, but I seem to remember it was happening with the same board which may be the duplicated one. But my memory could fail here. I don't know how to reproduce it now :)
* I don't have a Linux box at the moment but I can do it hopefully in the next days

**Comment by GSGBen** [comment]
Could you please try again now with `v4.0.3` and let me know if it's fixed.

**Comment by alranel** [comment]
I confirm this issue is fixed in v4.0.3. Thank you! :)
